File: binarytree.py
# Определим структуру для узла в дереве
class Node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

# Функция для вставки нового узла в дерево
def insert(root, data):
    if root is None:
        return Node(data)
    if data < root.data:
        root.left = insert(root.left, data)
    else:
        root.right = insert(root.right, data)
    return root

# Функция для выполнения обхода дерева по порядку
def inorder_traversal(root):
    if root:
        inorder_traversal(root.left)
        print(root.data, end=' ')
        inorder_traversal(root.right)

root = None
for value in [50, 30, 70, 20, 40, 60, 80]:
    root = insert(root, value)
    
# Обход по порядку
print("Обход по порядку: ")
inorder_traversal(root)
print()

File: BinaryTree.java
// Определим структуру для узла в дереве
class Node {
    int data;
    Node left, right;

    public Node(int item) {
        data = item;
        left = right = null;
    }
}

// // Определим структуру для дерева
public class BinaryTree {
    Node root;

    public BinaryTree() {
        root = null;
    }

    // Функция для вставки нового узла в дерево
    public Node insert(Node root, int data) {
        if (root == null) {
            return new Node(data);
        }
        if (data < root.data) {
            root.left = insert(root.left, data);
        } else {
            root.right = insert(root.right, data);
        }
        return root;
    }

    // Функция для выполнения обхода дерева по порядку
    public void inorderTraversal(Node root) {
        if (root != null) {
            inorderTraversal(root.left);
            System.out.print(root.data + " ");
            inorderTraversal(root.right);
        }
    }

    public static void main(String[] args) {
        BinaryTree tree = new BinaryTree();

        tree.root = tree.insert(tree.root, 50);
        tree.insert(tree.root, 30);
        tree.insert(tree.root, 70);
        tree.insert(tree.root, 20);
        tree.insert(tree.root, 40);
        tree.insert(tree.root, 60);
        tree.insert(tree.root, 80);

        // Обход по порядку:
        System.out.print("Inorder traversal: ");
        tree.inorderTraversal(tree.root);
        System.out.println();
    }
}
